src/base/perf_monitor.c

```diff
@@ -3011,6 +3022,22 @@ perfmon_initialize (int num_trans)
     }
 #endif
 
+  /* check proper ordering of statistics definitions:
+   *  - non-complex definitions come before the complex ones
+   *  - complex definitions all come at the end, just before PSTAT_COUNT
+   * needed because dump to file/buffer functions depend on this
+   * NOTE: this should actually be a unit test */
+  for (idx = 0; idx < PSTAT_COUNT && pstat_Metadata[idx].valtype != PSTAT_COMPLEX_VALUE; ++idx);
```

Comment:
What is the purpose of this for loop? I think this 'for' statement ends on this single line because of ';'. Is this correct what you intended? If it is, Plz explain more detail to me.

Reply:
Sorry. I understand your intention.

Reply:
Yes, it works because `idx` variable's scope is outside the loop and, thus, the compiler sees side effect and does not optimize it away (not even in release/optimized compilation mode).

Reply:
My suggestion is to move the check to a separate function and make it obvious that it is meant for debug by using preprocessor directive `if !defined (NDEBUG)`

Reply:
Done